File: NotificationSystem/NotificationSystem.Web/Controls_Add/ctrTroubleTicketReq_Add.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Globalization;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using Telerik.Web.UI;
using System.Data.SqlClient;
using NotificationSystem.NotificationSystem.Data.NotificationSystemTableAdapters;
using static NotificationSystem.NotificationSystem.Data.NotificationSystem;
using NotificationSystem.NotificationSystem.Data.Classes;
using System.Net.Http;

    public partial class ctrTroubleTicketReq_Add : System.Web.UI.UserControl
    {
        private int m_TroubleTicketNo = 0;

        public int TroubleTicketNo
        {
            get
            {
                return m_TroubleTicketNo;
            }
            set
            {
                m_TroubleTicketNo = value;
            }
        }

        public TextBox txtTroubleTicketList { get; set; }
        protected void Page_Load(object sender, EventArgs e)
        {
        if ((Page.IsPostBack))
        {
            try
            {
                if (Request.Form["ctl00$MainContent$ctrTroubleTicketReq_Add$btnAdd"] == "Add")
                    AddTroubleTicket();

                else if (Request.Form["ctl00$MainContent$ctrTroubleTicketReq_Search$btnSearch"] == "Search")
                {
                    clsNotificationSystem theNotificationSystem = new clsNotificationSystem();

                    TroubleTicketReqDataTable tblTroubleTicketReq = (TroubleTicketReqDataTable)theNotificationSystem.GetTroubleTicket(m_TroubleTicketNo);

                    {
                        var withBlock = tblTroubleTicketReq[0];
                        withBlock.AgentID = int.Parse(this.txtAgentID.Text); withBlock.CustomerID = int.Parse(this.txtcustomerid.Text); withBlock.TroubleTicketNo = int.Parse(this.txttroubleticketno.Text);
                        cmbStatus.Text = withBlock.Status;
                        cmbType.Text = withBlock.Type;
                        withBlock.RequestDate = DateTime.Now;
                        withBlock.DueDate = DateTime.Now;
                    }
                }

                PopulateControls();
            }
            catch (Exception ex)
            {
                clsNotificationSystem_Web SendError = new clsNotificationSystem_Web();
                string NotificationBody = ex.Message + "  " + ex.StackTrace;
                SendError.SendMailMessage(NotificationBody);
                Response.Redirect("ErrorPage.aspx", false);
            }
        }
    }

        public void AddTroubleTicket()
        {
            try
            {
                clsTroubleTicketReq thisTroubleTicket = new clsTroubleTicketReq();

                {
                    var withBlock = thisTroubleTicket;
                    if (txttroubleticketno.Text.Length == 0)

                        //if (pickRequestDate.SelectedDate == null)
                        //    withBlock.RequestDate = DateTime.Now;
                        //else
                        //    withBlock.RequestDate = (DateTime)pickRequestDate.SelectedDate;
                    if (cmbType.Text == string.Empty)
                        return;



                    withBlock.CustomerID = int.Parse(txtcustomerid.Text); withBlock.AgentID = int.Parse(txtAgentID.Text); withBlock.TroubleTicketNo = int.Parse(txttroubleticketno.Text);
                    cmbStatus.Text = withBlock.Status; 
                    cmbType.Text = withBlock.Type; 
                    withBlock.DueDate = DateTime.Now; 
                    withBlock.RequestDate = DateTime.Now;


            }
            try

                {

                    clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
                    theNotificationSystem.AddTroubleTicket(thisTroubleTicket);
                    lblResult.Text = "Trouble Ticket data has been added";

                }

                catch (SqlException ex)
                {
                    if (ex.Number == 2627)
                        lblResult.Text = "Ticket already exist!";
                    else
                        throw new ApplicationException(ex.Message);
                }

            }
            catch (Exception ex)
            {
                clsNotificationSystem_Web SendError = new clsNotificationSystem_Web();
                string NotificationBody = ex.Message + "  " + ex.StackTrace;
                SendError.SendMailMessage(NotificationBody);
                Response.Redirect("ErrorPage.aspx", false);
            }
        return;
        }

        public void ClearControls()
        {
            try
            {
            pickRequestDate.Clear();
            cmbStatus.Value = string.Empty;  
            cmbStatus.Text = string.Empty;
            cmbType.Value = string.Empty;
            cmbType.Text = string.Empty;
           
        }


            catch (Exception)
            {
                throw;
            }
        }

	private void PopulateControls()
	{
		clsNotificationSystem theNotificationSystem = new clsNotificationSystem();



        cmbStatus.DataSource = theNotificationSystem.GetStatusList();
        cmbStatus.DataTextField = "Status";
        cmbStatus.DataBind();



        cmbType.DataSource = theNotificationSystem.GetTypeList();
        cmbType.DataTextField = "Type";
        cmbType.DataBind();


    }
	protected void btnCancel_Click(object sender, EventArgs e)
        {
            Response.Redirect("TroubleTicketReq_Find.aspx", false);
        }
    }

File: NotificationSystem/NotificationSystem.Web/Controls_Add/ctrTroubleTicketReq_Add.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Data.SqlClient
Imports NotificationSystem.NotificationSystem.Data.NotificationSystem
Imports NotificationSystem.NotificationSystem.Data.Classes

Public Partial Class ctrTroubleTicketReq_Add
    Inherits UserControl
    Private m_TroubleTicketNo As Integer = 0

    Public Property TroubleTicketNo As Integer
        Get
            Return m_TroubleTicketNo
        End Get
        Set(value As Integer)
            m_TroubleTicketNo = value
        End Set
    End Property

    Public Property txtTroubleTicketList As TextBox
    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        PopulateControls()
        If Not IsPostBack Then
            txtcustomerid.Text = TroubleTicketNo.ToString()
            txtcustomerid.Enabled = False
            txtAgentID.Text = "0"
            txttroubleticketno.Text = "0"
            cmbStatus.Text = ""
            cmbType.Text = ""
        End If
    End Sub

    Public Sub AddTroubleTicket()
        Try
            Dim thisTroubleTicket As New clsTroubleTicketReq


            With thisTroubleTicket

                If pickRequestDate.SelectedDate Is Nothing Then .RequestDate = Date.Now Else .RequestDate = CDate(pickRequestDate.SelectedDate)


                If String.IsNullOrEmpty(cmbType.Text) Then Return

                .CustomerID = Integer.Parse(txtcustomerid.Text)
                .AgentID = Integer.Parse(txtAgentID.Text)
                .TroubleTicketNo = Integer.Parse(txttroubleticketno.Text)
                .Status = cmbStatus.Text
                .Type = cmbType.Text
                .DueDate = Date.Now

                If txttroubleticketno.Text.Length = 0 Then
                    ' Handle empty TroubleTicketNo if necessary
                End If
            End With
            Dim theNotificationSystem As New clsNotificationSystem
            theNotificationSystem.AddTroubleTicket(thisTroubleTicket)
                lblResult.Text = "Trouble Ticket data has been added"


        Catch ex As Exception
            Dim SendError As clsNotificationSystem_Web = New clsNotificationSystem_Web()
            Dim NotificationBody = ex.Message & "  " & ex.StackTrace
            SendError.SendMailMessage(NotificationBody)
            Response.Redirect("ErrorPage.aspx", False)
        End Try

    End Sub

    Public Sub ClearControls()
        Try
            'pickRequestDate.Clear();
            cmbStatus.Value = String.Empty
            cmbStatus.Text = String.Empty
            cmbType.Value = String.Empty
            cmbType.Text = String.Empty


        Catch __unusedException1__ As Exception
            Throw
        End Try
    End Sub

    Private Sub PopulateControls()
        Dim theNotificationSystem As clsNotificationSystem = New clsNotificationSystem()



        cmbStatus.DataSource = theNotificationSystem.GetStatusList()
        cmbStatus.DataTextField = "Status"
        cmbStatus.DataValueField = "Status"
        cmbStatus.DataBind()



        cmbType.DataSource = theNotificationSystem.GetTypeList()
        cmbType.DataTextField = "Type"
        cmbType.DataValueField = "Type"
        cmbType.DataBind()


    End Sub
    Protected Sub btnCancel_Click(sender As Object, e As EventArgs)
        Response.Redirect("TroubleTicketReq_Find.aspx", False)
    End Sub
End Class
